components/mailgun/sources/new-bounce/new-bounce.js

```diff
@@ -0,0 +1,50 @@
+const common = require("../common-webhook");
+const { mailgun } = common.props;
+
+module.exports = {
+  ...common,
+  key: "new-bounce",
+  name: "New bounce",
+  description:
+    "Emit an event when the email recipient who could not be reached.",
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
+    ...common.props,
+    domain: { propDefinition: [mailgun, "domain"] },
+  },
+  methods: {
+    ...common.methods,
+    getEventName() {
+      return ["bounce"];
+    },
+    getEventType() {
```

Comment:
These need to return arrays in each of the components, like `["bounced"]`. The way you use it in `emitEvent`:

```javascript
const eventTypes = this.getEventType();
if (eventTypes.includes(eventPayload.event)) {
```

input like "bounce" or "bou" will also return true, since that's the way the `includes` method works for strings. If you return an array, `includes` demands that the argument be an element of the array, which means only "bounces" will match.

Reply:
ah, I see. In the examples that I saw, I thought of using the arrays as in `["bounced"]`  and thought it was because the related service being used to create the component for had more than 1 type. I think it was Github the sample I saw. 

The change was made to return arrays.